apm-agent-plugins/apm-micrometer-plugin/src/main/java/co/elastic/apm/agent/micrometer/MicrometerMeterRegistrySerializer.java

```diff
@@ -199,28 +210,40 @@ private static void serializeValue(Meter.Id id, String suffix, long value, JsonW
      * @param value    meter value
      * @param hasValue whether a value has already been written
      * @param jw       writer
+     * @param replaceBuilder
+     * @param dedotMetricName
      * @return true if a value has been written before, including this one; false otherwise
      */
-    private static boolean serializeValue(Meter.Id id, double value, boolean hasValue, JsonWriter jw) {
+    private static boolean serializeValue(Meter.Id id, double value, boolean hasValue, JsonWriter jw, StringBuilder replaceBuilder, boolean dedotMetricName) {
         if (isValidValue(value)) {
             if (hasValue) jw.writeByte(JsonWriter.COMMA);
-            serializeValue(id, "", value, jw);
+            serializeValue(id, "", value, jw, replaceBuilder, dedotMetricName);
             return true;
         }
         return hasValue;
     }
 
-    private static void serializeValue(Meter.Id id, String suffix, double value, JsonWriter jw) {
-        serializeValueStart(id.getName(), suffix, jw);
+    private static void serializeValue(Meter.Id id, String suffix, double value, JsonWriter jw, StringBuilder replaceBuilder, boolean dedotMetricName) {
+        serializeValueStart(id.getName(), suffix, jw, replaceBuilder, dedotMetricName);
         NumberConverter.serialize(value, jw);
         jw.writeByte(JsonWriter.OBJECT_END);
     }
 
-    private static void serializeValueStart(String key, String suffix, JsonWriter jw) {
-        jw.writeByte(JsonWriter.QUOTE);
-        jw.writeAscii(key);
-        jw.writeAscii(suffix);
-        jw.writeByte(JsonWriter.QUOTE);
+    private static void serializeValueStart(String key, String suffix, JsonWriter jw, StringBuilder replaceBuilder, boolean dedotMetricName) {
+        replaceBuilder.setLength(0);
+        if (dedotMetricName) {
+            DslJsonSerializer.sanitizeLabelKey(key, replaceBuilder);
```

Comment:
[minor] maybe rename method as it's now not only used for label keys, but more keys in general.